ide-support/revliburl.livecodescript

```diff
@@ -1760,50 +1760,58 @@ function ulNormaliseEncoding pEncoding
 end ulNormaliseEncoding
 
 -- Convert the data according to the encoding specified
-command ulDecodeData pUrl
-   
-   -- MM-2014-04-08: [[ Bug 12006 ]] Ignore the content type encoding for pre 7.0 versions of LiveCode.
-   if the buildNumber < 10000 then
-      return empty
-   end if
-   
-   local tEncoding
-   put empty into tEncoding
-   
-   local tHeadSize
-   --   Only look for the encoding in the head part
-   put offset("</head>", laData[pUrl]) into tHeadSize
-   
-   if tHeadSize is 0 then exit "ulDecodeData"
-   
-   local tHead
-   put char 1 to tHeadSize of laData[pUrl] into tHead
-   
-   local tCharsetRegex
-   put "<meta .*charset=" & quote & "?([a-zA-Z0-9 -]+)" & quote & ".*>" into tCharsetRegex
-   
-   local tCharset
-   if matchtext(tHead, tCharsetRegex, tCharset) then      
-      switch ulNormaliseEncoding(tCharset)
-         case "utf8"            
-            put "UTF8" into tEncoding
-            break
-         case "utf16"
-            put "UTF16" into tEncoding
-            break
-         case "iso88591"
-            put "ANSI" into tEncoding
-            break
-      end switch
-      
-      if tEncoding is not empty then
-         local tDecoded
-         put uniEncode(laData[pUrl], tEncoding) into tDecoded
-         put uniDecode(tDecoded) into laData[pUrl]   
-      end if
-   end if
+command ulDecodeData pHeaders, @pData
+  local tEncoding, tCharset, tLineNo
+
+  put empty into tEncoding
```

Comment:
One option would be to change this line to `put CheckForBOM(pData) into tEncoding` and then check to see if it is not empty.  If not, trim the necessary bytes from pData and skip the remainder of the checks and just decode.  Either UTF32 is 4 bytes, either UTF16 is 2 bytes, and UTF8 is 3 bytes.